spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/StreamCommands.java

```diff
@@ -179,45 +197,38 @@ public String deployStream(
 			propertiesToUse = DeploymentPropertiesUtils.convert(props);
 			break;
 		case -1: // Neither option specified
-			propertiesToUse = (useSkipper) ? new HashMap<>(1) : Collections.<String, String>emptyMap();
+			propertiesToUse = new HashMap<>(1);
 			break;
 		default:
 			throw new AssertionError();
 		}
-		if (useSkipper) {
-			propertiesToUse.put(SKIPPER_ENABLED_PROPERTY_KEY, "true");
-			propertiesToUse.put(SKIPPER_PACKAGE_NAME , StringUtils.hasText(packageName) ? packageName : name);
-			Assert.isTrue(StringUtils.hasText(packageVersion), "Package version must be set when using Skipper.");
-			propertiesToUse.put(SKIPPER_PACKAGE_VERSION, packageVersion);
-			if (StringUtils.hasText(platformName)) {
-				propertiesToUse.put(SKIPPER_PLATFORM_NAME, platformName);
-			}
-			if (StringUtils.hasText(repoName)) {
-				propertiesToUse.put(SKIPPER_REPO_NAME, repoName);
-			}
-		}
-		else {
-			Assert.isTrue(platformName == null, "Platform name can only be specified when deploying the stream using "
-							+ "Skipper");
-			Assert.isTrue(repoName == null, "Repository name can only be specified when deploying the stream using "
-					+ "Skipper");
-			Assert.isTrue(packageName == null, "Package name can only be specified when deploying the stream using "
-					+ "Skipper");
-			Assert.isTrue(packageVersion == null, "Package version can only be specified when deploying the stream "
-					+ "using Skipper");
-		}
+		return propertiesToUse;
+	}
+
+	@CliCommand(value = DEPLOY_STREAM, help = "Deploy a previously created stream")
+	public String deployStream(
+			@CliOption(key = { "",
+					"name" }, help = "the name of the stream to deploy", mandatory = true, optionContext = "existing-stream disable-string-converter") String name,
+			@CliOption(key = {
+					PROPERTIES_OPTION }, help = "the properties for this deployment") String deploymentProperties,
+			@CliOption(key = {
+					PROPERTIES_FILE_OPTION }, help = "the properties for this deployment (as a File)") File propertiesFile)
+			throws IOException {
+		int which = Assertions.atMostOneOf(PROPERTIES_OPTION, deploymentProperties, PROPERTIES_FILE_OPTION,
+				propertiesFile);
+		Map<String, String> propertiesToUse = getDeploymentProperties(deploymentProperties, propertiesFile, which);
 		streamOperations().deploy(name, propertiesToUse);
 		return String.format("Deployment request has been sent for stream '%s'", name);
 	}
 
-	@CliCommand(value = UPDATE_STREAM, help = "Update a previously created stream")
+	@CliCommand(value = STREAM_SKIPPER_UPDATE, help = "Update a previously created stream")
```

Comment:
Albeit this is a skipper specific command, I think it might be useful to add "... using Skipper" for consistency. 

It shows like this now:

```
dataflow:>help stream skipper
* stream skipper deploy - Deploy a previously created stream using Skipper
* stream skipper update - Update a previously created stream
```

Reply:
I can add that with merge